pkg/apis/pipeline/v1alpha1/cluster_resource.go

```diff
@@ -62,11 +62,12 @@ func NewClusterResource(kubeconfigWriterImage string, r *PipelineResource) (*Clu
 	clusterResource := ClusterResource{
 		Type:                  r.Spec.Type,
 		KubeconfigWriterImage: kubeconfigWriterImage,
+		Name:                  r.Name,
```

Comment:
If I understand this correctly, this uses the resource name as cluster name. Unfortunately this is not OK - which is the reason the name parameter was added - because the resource name must be a DNS valid name, while the cluster name may be not a valid DNS one e.g. my_cluster is a valid cluster name, but not a valid resource name.

Reply:
Ah thanks for the explanation. Can you explain why the cluster name matters in your case?

Leaving the name parameter would be fine I think as long as we write to the /workspace/resource-name path instead of the /workspace/cluster-name-param path.

I can't actually see the use case for cluster name though - it's only used inside the .kubeconfig AFAICT, and can be literally anything.

Reply:
So, indeed it doesn't seem to matter at all, sorry about my bad review :(